converters/fromZigbee.js

```diff
@@ -3409,6 +3409,17 @@ const converters = {
             return {moving: msg.data['acceleration'] === 1 ? true : false};
         },
     },
+    smartthings_accelerometer: {
+        cluster: 'manuSpecificSamsungAccelerometer',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            return {
+                x_axis: msg.data['x_axis'],
```

Comment:
I think you might need to check for the existence of the `x_axis`, `y_axis` and `z_axis` attributes in the Zigbee packet before you include it in the returned object. That could be causing the odd quirk that you mentioned. I'm not a JS guy, reference to a nonexistent attribute this would throw an exception in Python but in JS maybe it returns `null` which is converted somewhere to 0.

Reply:
Hmmm, ya sounds about right. In JS this will return `undefined` and maybe somewhere upstream keys with undefined values are being removed.

Reply:
In case that e.g. `x_axis` is not defined it should not be returned. If e.g. first a message with `acceleration` and `x_axis` is received and next a message with only `acceleration` the `x_axis` property will be cleared (while it should be preserved.

Changing to:

```js
return {
    ...(msg.data.hasOwnProperty('acceleration') && {moving: msg.data['acceleration'] === 1}),
    ...(msg.data.hasOwnProperty('x_axis') && {x_axis: msg.data['x_axis']}),
    ...(msg.data.hasOwnProperty('y_axis') && {y_axis: msg.data['y_axis']}),
    ...(msg.data.hasOwnProperty('z_axis') && {z_axis: msg.data['z_axis']}),
}
```

should solve it.

Reply:
You might also want to look here at the [SmartThings DTH code](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy#L222), it does this mangling with the received `x/y/z_axis` attributes when manufacturer = `SmartThings` (the only model supported by the DTH with this manufacturer is `multiv4`):
```
xyzResults.x = z
xyzResults.y = y
xyzResults.z = -x
```

Reply:
Thx.
Do i need to worry about the manufacturer available to me in this location to implement the same check that we have or should I just go with the "SmartThings" case?

Reply:
The SmartThings DTH seems to be written to cover a single product ("SmartThings multi-sensor") that have multiple hardware revisions that are sometimes made by completely different OEM manufacturers. In z2m, each hardware revision has a separate device profile that can be set up to trigger either the same or different converters. So I'd suggest that you just code the case specific for `multiv4`. If someone decides to add accelerator support to any of the other devices supported by that DTH, then they can either modify the converter you wrote in this PR to handle the other cases, or (more likely as it's relatively simple) write a separate converter that mangles `x/y/z-axis` as required for that device.

Reply:
> xyzResults.x = z
> xyzResults.y = y
> xyzResults.z = -x

This explains why I had to change all my orientation logic when I switched this sensor over from SmartThings to ZHA in Home Assistant.